src/ClientHandle.cpp

```diff
@@ -1630,21 +1638,15 @@ void cClientHandle::HandleUseEntity(UInt32 a_TargetEntityID, bool a_IsLeftClick)
 					return true;
 				}
 			}
-			a_Entity->TakeDamage(*m_Attacker);
-			return false;
+			a_Entity->TakeDamage(*m_Me);
+			m_Me->AddFoodExhaustion(0.3);
```

Comment:
By the way, previously this line never executed and there was no food exaustion when attacking mobs.